runtime/compiler/compile/J9SymbolReferenceTable.cpp

```diff
@@ -842,6 +845,86 @@ J9::SymbolReferenceTable::findOrFabricateShadowSymbol(
    return symRef;
    }
 
+TR::SymbolReference *
+J9::SymbolReferenceTable::findFlattenedArrayElementFieldShadow(
+   FlattenedArrayElementFieldShadowKey key,
+   bool isVolatile,
+   bool isPrivate,
+   bool isFinal)
+   {
+   const auto entry = _flattenedArrayElementFieldShadows.find(key);
+   if (entry == _flattenedArrayElementFieldShadows.end())
+      return NULL;
+
+   TR::SymbolReference *symRef = entry->second;
+   int32_t refNum = symRef->getReferenceNumber();
+   TR::Symbol *sym = symRef->getSymbol();
+
+   TR_ASSERT_FATAL(sym->isVolatile() || !isVolatile, "expecting volatile symref but found non-volatile symref #%d\n", refNum);
+   TR_ASSERT_FATAL(!sym->isPrivate() || isPrivate, "expecting non-private symref but found private symref #%d\n", refNum);
+   TR_ASSERT_FATAL(!sym->isFinal() || isFinal, "expecting non-final symref but found final symref #%d\n", refNum);
+
+   return symRef;
+   }
+
+TR::SymbolReference *
+J9::SymbolReferenceTable::findOrFabricateFlattenedArrayShadowSymbol(
+   TR_OpaqueClassBlock *arrayComponentClass,
+   TR::DataType type,
+   uint32_t fieldOffset,
+   bool isVolatile,
+   bool isPrivate,
+   bool isFinal,
+   const char *fieldName,
+   const char *fieldSignature)
+   {
+   int32_t flattenedFieldOffset = (int32_t)fieldOffset - (int32_t)TR::Compiler->om.objectHeaderSizeInBytes();
+   TR_ASSERT_FATAL(flattenedFieldOffset >= 0, "flattenedFieldOffset %d is invalid: fieldOffset %u objectHeaderSizeInBytes %u", fieldOffset, TR::Compiler->om.objectHeaderSizeInBytes());
```

Comment:
Did you mean to pass `flattenedFieldOffset` for the first `%d`?

Also, `objectHeaderSizeInBytes()` is a `uintptr_t`, which on 64-bit will be larger than expected for `%u`, and the formatting will get confused. You can either cast it to `unsigned int` (or `uint32_t`, since we seem to generally assume that `int` is 32-bit), or use `OMR_PRIuPTR`

Reply:
> Did you mean to pass flattenedFieldOffset for the first %d?

I mean to pass `fieldOffset` to `%d` if `flattenedFieldOffset < 0` so we know why the calculated `flattenedFieldOffset` is wrong. 

>You can either cast it to unsigned int (or uint32_t, since we seem to generally assume that int is 32-bit), or use OMR_PRIuPTR

I will use `OMR_PRIuPTR`

Reply:
Addressed in f4541ce